### ISSUE TITLE
Interface ConfigurationOptions is missing newer Bugsnag options

### ISSUE THREAD

**Issue by my-name-was-already-taken** [bugsnag/bugsnag-node#130]
It seems that the TypeScript interface `ConfigurationOptions` is missing the following properties / new configuration options:

```
autoNotifyUncaught?: boolean;
autoNotifyUnhandledRejection?: boolean;
apiKey?: string;
autoCaptureSessions?: boolean;
sessionEndpoint?: string;
beforeNotifyCallbacks?: any;
```

... whereas `beforeNotifyCallbacks` and the already defined `logger` could maybe even be more strict (e.g. sth like `Array<(notification: any, error: any) => any>` or if you'd argue that the lack of typing for that means no breaking change with non-any-types, then Notification / Error and => boolean).

**Comment by bengourley** [comment]
Hi @my-name-was-already-taken (great name). Sorry for the delay in getting back to you. I'll schedule getting this fixed, and also look at adding some automate QA to ensure the types are complete/accurate for any future changes. Thanks for your patience!

**Comment by ajsharp** [comment]
@bengourley What's the status of this? I can knock this out quickly if you're open to a PR.

**Comment by bengourley** [comment]
Hi @ajsharp, we would gladly receive a PR! This work is scheduled but it's lower priority than other things we have on the table right now, so is unlikely to be visited in the immediate future. If you make a PR I'll expedite some time to get it reviewed/merged though. Cheers

**Comment by bengourley** [comment]
Fixed via #144.
